docs/blog/_posts/2018-07-06-ninth-dotty-milestone-release.md

```diff
@@ -0,0 +1,163 @@
+be a big step towards realizing the full potential of these ideas. Its main objectives are to
+
+- become more opinionated by promoting programming idioms we found to work well,
+- simplify where possible,
+- eliminate inconsistencies and surprising behaviors,
+- build on strong foundations to ensure the design hangs well together,
+- consolidate language constructs to improve the language’s consistency, safety, ergonomics, and performance.
+
+You can learn more about Dotty on our [website](https://dotty.epfl.ch).
+
+<!--more-->
+
+This is our ninth scheduled release according to our [6-week release schedule](https://dotty.epfl.ch/docs/usage/version-numbers.html).
+The [previous technology preview](https://github.com/lampepfl/dotty/releases/tag/0.8.0-RC1) added
+support for sbt 1, introduced improved unchecked warnings and improved SAM type support.
+
+## What’s new in the 0.9.0-RC1 technology preview?
+
+### Improved REPL [#4680](https://github.com/lampepfl/dotty/pull/4680)
+The REPL now uses [JLine 3](https://github.com/jline/jline3) under the hood which improves on
+many aspects such as auto-completions and multi-line editing. The REPL now also works on Windows!
+
+### Documentation support in the IDE [#4461](https://github.com/lampepfl/dotty/pull/4461), [#4648](https://github.com/lampepfl/dotty/pull/4648)
+The Dotty IDE will now display documentation on code hover for symbols that were previously
+compiled by the Dotty compiler. In the future, we plan to let users query the documentation
+in the REPL as well.
+
+### Drop requirement that implicit functions must be non-empty [#4549](https://github.com/lampepfl/dotty/pull/4549)
+We decided to remove an arbitrary restriction that implicit and functions must be non-empty.
+We can now write:
+```scala
+type IntProducer = implicit () => Int
+
+def prod1: IntProducer = 1
+val prod2: IntProducer = 2
+```
+
+An interesting observation is that by-name parameters can now be encoded as implicit function types:
+```scala
+def timed[T](op: => T) = ...
+def timed[T](op: implicit () => T) = ...
+
+timed {
+  fetch(url)
+}
+```
+
+Both definitions above are equivalent.
+
+### Emit feature warnings for implicit conversions [#4229](https://github.com/lampepfl/dotty/pull/4229)
+Implicit conversions are easily the most misused feature in Scala. We now emit feature warnings
+when encountering an implicit conversion definition, just like Scala 2 does.
+
+In addition, we also emit a feature warning when an implicit conversion is used,
+unless the conversion is an implicit class, or otherwise co-defined with the type
+to which it converts, or the conversion is predefined in `scala.Predef` or is the
+`scala.reflect.Selectable.reflectiveSelect` conversion (we might extend this to more conversions).
+
+### Optimise s and raw interpolators [#3961](https://github.com/lampepfl/dotty/pull/4229)
+`s` and `raw` string interpolators were known to be slower than their not type-safe counterparts:
```

Comment:
It seems both are equally type-safe. I would say "the non-interpolated code".

Reply:
In the  example below I agree, but that's not the point here. Here is another example that I think illustrate "type safety":
```scala
scala> val x = 1; val y = 2
x: Int = 1
y: Int = 2

scala> "Hello " + x + y
res0: String = Hello 12

scala> x + y + " Hello"
res1: String = 3 Hello

scala> s"Hello $x$y"
res2: String = Hello 12

scala> s"$x$y Hello"
res3: String = 12 Hello
```